docs/ref/forms/api.txt

```diff
@@ -830,107 +830,228 @@ The field-specific output honors the form object's ``auto_id`` setting::
+    method, or ``None`` if it wasn't given::
 
-    >>> f = ContactForm(data)
-    >>> print(f['message'].label_tag())
-    <label for="id_message">Message:</label>
+        >>> unbound_form = ContactForm()
+        >>> print(unbound_form['subject'].data)
+        None
+        >>> bound_form = ContactForm(data={'subject': 'My Subject'})
+        >>> print(bound_form['subject'].data)
+        My Subject
 
-Optionally, you can provide the ``contents`` parameter which will replace the
-auto-generated label tag. An optional ``attrs`` dictionary may contain
-additional attributes for the ``<label>`` tag.
+.. attribute:: BoundField.errors
 
-The HTML that's generated includes the form's
-:attr:`~django.forms.Form.label_suffix` (a colon, by default) or, if set, the
-current field's :attr:`~django.forms.Field.label_suffix`. The optional
-``label_suffix`` parameter allows you to override any previously set
-suffix. For example, you can use an empty string to hide the label on selected
-fields. If you need to do this in a template, you could write a custom
-filter to allow passing parameters to ``label_tag``.
+    A list-like object that is displayed as an HTML ``<ul class="errorlist">``
+    when printed::
 
-.. versionchanged:: 1.8
+        >>> data = {'subject': 'hi', 'message': '', 'sender': '', 'cc_myself': ''}
+        >>> f = ContactForm(data, auto_id=False)
+        >>> print(f['message'])
+        <input type="text" name="message" />
+        >>> f['message'].errors
+        ['This field is required.']
+        >>> print(f['message'].errors)
+        <ul class="errorlist"><li>This field is required.</li></ul>
+        >>> f['subject'].errors
+        []
+        >>> print(f['subject'].errors)
 
-    The label includes :attr:`~Form.required_css_class` if applicable.
+        >>> str(f['subject'].errors)
+        ''
 
-.. method:: BoundField.css_classes()
+.. attribute:: BoundField.field
 
-When you use Django's rendering shortcuts, CSS classes are used to
-indicate required form fields or fields that contain errors. If you're
-manually rendering a form, you can access these CSS classes using the
-``css_classes`` method::
+    The :class:`~django.forms.Field` instance from the form class that this
+    :class:`~django.forms.BoundField` wraps.
 
-    >>> f = ContactForm(data)
-    >>> f['message'].css_classes()
-    'required'
+.. attribute:: BoundField.form
 
-If you want to provide some additional classes in addition to the
-error and required classes that may be required, you can provide
-those classes as an argument::
+    The :class:`~django.forms.Form` instance this :class:`~django.forms.BoundField`
+    is bound to.
 
-    >>> f = ContactForm(data)
-    >>> f['message'].css_classes('foo bar')
-    'foo bar required'
+.. attribute:: BoundField.help_text
 
-.. method:: BoundField.value()
+    :attr:`~django.forms.Field.help_text` of the field.
+
+.. attribute:: BoundField.html_initial_id
```

Comment:
I think it depends on the outcome of the decision about how extensible this API is meant to be. Personally, I see BoundField as an API that's largely meant for use in templates, so documenting the methods that aren't meant to be called from there (e.g. most of the methods with arguments) doesn't make much sense to me. If we open things to allow substituting your own BoundField class, then this might change.